File: solutions/zig/isogram/1/isogram.zig
const std = @import("std");

pub fn isIsogram(str: []const u8) bool {
    var counts = [_]u8{0} ** 26;
    for (str) |c| {
        const char = std.ascii.toUpper(c);
        if (char < 'A' or char > 'Z') {
            continue;
        }
        const i = char - 'A';
        if (counts[i] > 0) {
            return false;
        }
        counts[i] += 1;
    }
    return true;
}

File: solutions/go/isogram/1/isogram.go
package isogram

import "strings"

func IsIsogram(word string) bool {
	counts := make(map[rune]bool)
	for _, char := range strings.ToUpper(word) {
		if char < 'A' || char > 'Z' {
			continue
		}
		if counts[char] {
			return false
		}
		counts[char] = true
	}
	return true
}
